bee-node/bee-node/src/workers.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{convert::Infallible, net::SocketAddr, time::Duration};
+
+use async_trait::async_trait;
+use bee_metrics::{metrics::ProcessMetrics, Registry};
+use bee_runtime::{node::Node, shutdown_stream::ShutdownStream, worker::Worker};
+use futures::StreamExt;
+use log::info;
+use serde::Deserialize;
+use tokio::time::interval;
+use tokio_stream::wrappers::IntervalStream;
+
+const PROCESS_METRICS_UPDATE_INTERVAL: Duration = Duration::from_secs(5);
+const DEFAULT_BIND_ADDRESS: &str = "0.0.0.0:3030";
+
+#[derive(Default, Deserialize, PartialEq)]
+#[must_use]
+pub struct MetricsConfigBuilder {
+    #[serde(skip)]
+    pid: Option<u32>,
+    #[serde(alias = "bindAddress")]
+    bind_address: Option<SocketAddr>,
+}
+
+impl MetricsConfigBuilder {
+    pub fn with_pid(mut self, pid: u32) -> Self {
+        self.pid = Some(pid);
+        self
+    }
+
+    pub fn finish(self) -> MetricsConfig {
+        MetricsConfig {
+            pid: self.pid.unwrap(),
```

Comment:
Can we maybe unwrap or get the pid ? This can panic

Reply:
I'm not sure if we can guarantee that we are getting the PID of the process we want that way. The only reason this is an option is so we can keep `MetricsConfig` in a single structure and it can be deserialized. I can change that and make it so

```rust
MetricsConfigBuilder {
    pid: u32,
    rest: DeserializableMetricsConfigBuilder,
}
```

Reply:
Not sure to follow, the field is `#[serde(skip)]`

Reply:
Oh that's a good idea.